test/upgrade/prober/prober.go

```diff
@@ -114,11 +114,19 @@ func (p *prober) servingClient() resources.ServingClient {
 	}
 }
 
-func (p *prober) ReportError(t *testing.T, err error) {
-	if p.config.FailOnMissingEvents {
-		t.Error(err)
-	} else {
-		p.log.Warnf("Silenced FAIL: %v", err)
+func (p *prober) ReportErrors(t *testing.T, errors []error) {
+	for _, err := range errors {
+		if p.config.FailOnMissingEvents {
+			t.Error(err)
+		} else {
+			p.log.Warnf("Silenced FAIL: %v", err)
+		}
+	}
+	if len(errors) > 0 && !p.config.FailOnMissingEvents {
```

Comment:
Would it be better to have a different config option for FailOnErrors? Right now it has double meaning and without reading the code it's hard to know that setting that flag results on this behaviour?
Maybe it would be better to either have:
 - FailOnErrors _and_ FailOnMissingEvents config options
 - rename FailOnMissingEvents to FailOnErrors?

I think it's fine to get this in to make progress, just some thoughts.

Reply:
I've missed that. I've should have renamed that to `FailOnErrors`